### PR TITLE
AKS template cleanup

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**: 
- remove the multi-tenancy AKS flavor: the default for all clusters is to use multi-tenancy (AzureClusterIdentity), so the separate flavor is no longer relevant. In fact, the "aks" flavor is already using multi-tenancy.
- added a patch for the third agentpool to the test dir since it only meant for testing and should not be part of the AKS flavor we publish as a release asset
- removed the orphaned AAD AKS flavor which has no flavor directory (and is only showing a specific feature, since we want to get away from the "one feature, one flavor" model ref #1456)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Removed aks-multi-tenancy flavor and the aks-aad flavors
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/retest

flake is known #2120

**Comment by CecileRobertMichon** [comment]
same issue

/retest

File: templates/test/ci/cluster-template-prow-aks.yaml [hunk 1]
```diff
@@ -114,6 +113,22 @@ spec:
     key: type
     value: shared
 ---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureClusterIdentity
+metadata:
+  labels:
+    clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
```
Comment (jackfrancis) [inline templates/test/ci/cluster-template-prow-aks.yaml:120]: What's the significance of this additional label?

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
/hold cancel

rebased on top of #2095

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
/retest

```
"error"="failed to reconcile cluster services: failed to reconcile route table: failed to create resource capz-qadgta/capz-qadgta-node-routetable (service: routetables): network.RouteTablesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code=\"ResourceGroupBeingDeleted\" 
```

Looks like we had either a naming collision or something/someone deleted the RG out of band during the test

**Pull request merged**